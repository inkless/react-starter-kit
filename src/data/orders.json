[
  {
    "id": "9H4whD",
    "name": "Banana Split",
    "destination": "185 Culver Blvd, Playa Del Rey, CA 90293",
    "dispatchTime": 119
  },
  {
    "id": "EUtSqE",
    "name": "McFlury",
    "destination": "13455 Maxella Ave, Marina Del Rey, CA 90292",
    "dispatchTime": 98
  },
  {
    "id": "uoierx",
    "name": "Acai Bowl",
    "destination": "4720 Admiralty Way, Marina Del Rey, CA 90292",
    "dispatchTime": 35
  },
  {
    "id": "oP6QHd",
    "name": "Yogurt",
    "destination": "220 Rose Ave, Venice, CA 90291",
    "dispatchTime": 105
  },
  {
    "id": "aS4EFD",
    "name": "Chocolate Gelato",
    "destination": "2323 Wilshire Blvd, Santa Monica, CA 90403",
    "dispatchTime": 73
  },
  {
    "id": "4WVKZu",
    "name": "Cobb Salad",
    "destination": "1630 Ocean Park Blvd, Santa Monica, CA 90405",
    "dispatchTime": 13
  },
  {
    "id": "5sSNvC",
    "name": "Cottage Cheese",
    "destination": "1416 4th St, Santa Monica, CA 90401",
    "dispatchTime": 3
  },
  {
    "id": "Fv8AdV",
    "name": "Coke",
    "destination": "8115 W Manchester Ave, Playa Del Rey, CA 90293",
    "dispatchTime": 119
  },
  {
    "id": "WFG4B3",
    "name": "Snow Cone",
    "destination": "2055 Sawtelle Blvd, Los Angeles, CA 90025",
    "dispatchTime": 49
  },
  {
    "id": "woqBZv",
    "name": "Pad See Ew",
    "destination": "2906 Main St, Santa Monica, CA 90405",
    "dispatchTime": 94
  },
  {
    "id": "ZEUn6r",
    "name": "Chunky Monkey",
    "destination": "400 World Way, Los Angeles, CA 90045",
    "dispatchTime": 24
  },
  {
    "id": "tG7Mtb",
    "name": "Beef Stew",
    "destination": "2208 Sawtelle Blvd, Los Angeles, CA 90064",
    "dispatchTime": 16
  },
  {
    "id": "5kyaoG",
    "name": "Cheese",
    "destination": "1303 Westwood Blvd, Los Angeles, CA 90024",
    "dispatchTime": 63
  },
  {
    "id": "2YVKCU",
    "name": "Spinach Omelet",
    "destination": "1522 Abbott Kinney Blvd, Los Angeles, CA 90291",
    "dispatchTime": 78
  },
  {
    "id": "z57hNM",
    "name": "Beef Hash",
    "destination": "10117 Washington Blvd, Culver City, CA 90232",
    "dispatchTime": 24
  },
  {
    "id": "7FNaQi",
    "name": "Pork Chop",
    "destination": "1716 S Sepulveda Blvd, Los Angeles, CA 90025",
    "dispatchTime": 93
  },
  {
    "id": "EJyQaE",
    "name": "Kale Salad",
    "destination": "9409 Venice Blvd, Culver City, CA 90232",
    "dispatchTime": 110
  },
  {
    "id": "n5byZt",
    "name": "Fresh Fruit",
    "destination": "13455 Maxella Ave, Ste 250, Marina Del Rey, CA 90292",
    "dispatchTime": 66
  },
  {
    "id": "8yJnGe",
    "name": "Cranberry Salad",
    "destination": "101 Santa Monica Blvd, Santa Monica, CA 90401",
    "dispatchTime": 109
  },
  {
    "id": "tr5K86",
    "name": "Fudge Ice Cream Cake",
    "destination": "1321 3rd St Promenade, Santa Monica, CA 90401",
    "dispatchTime": 54
  },
  {
    "id": "UWkU5v",
    "name": "Mint Chocolate Ice Cream",
    "destination": "267 N Canon Dr, Beverly Hills, CA 90210",
    "dispatchTime": 39
  },
  {
    "id": "2ndwz7",
    "name": "Vegan Pizza",
    "destination": "3306 Lincoln Blvd, Santa Monica, CA 90405",
    "dispatchTime": 40
  },
  {
    "id": "7VD4xy",
    "name": "Orange Chicken",
    "destination": "3918 Van Buren Pl, Culver City, CA 90232",
    "dispatchTime": 112
  },
  {
    "id": "3hWiEm",
    "name": "MeatLoaf",
    "destination": "11777 San Vicente Blvd., Los Angeles, CA 90049",
    "dispatchTime": 26
  },
  {
    "id": "PQnWkG",
    "name": "Milk",
    "destination": "2310 Pico Blvd, Santa Monica, CA 90405",
    "dispatchTime": 112
  },
  {
    "id": "AinKVi",
    "name": "Pastrami Sandwich",
    "destination": "10113 Washington Blvd, Culver City, CA 90232",
    "dispatchTime": 120
  },
  {
    "id": "VrXRDR",
    "name": "Arugula",
    "destination": "2911 Main St, Santa Monica, CA 90405",
    "dispatchTime": 41
  },
  {
    "id": "sKar6C",
    "name": "Pickles",
    "destination": "210 Wilshire Blvd, Santa Monica, Santa Monica, CA 90401",
    "dispatchTime": 109
  },
  {
    "id": "7Ur6qT",
    "name": "Chicken",
    "destination": "193 Culver Blvd, Playa Del Rey, CA 90293",
    "dispatchTime": 47
  },
  {
    "id": "RoYGTg",
    "name": "Cookie Dough",
    "destination": "1323 Montana Ave, Santa Monica, CA 90403",
    "dispatchTime": 38
  },
  {
    "id": "swfNsi",
    "name": "Hamburger",
    "destination": "4342 Sepulveda Blvd, Culver City, CA 90230",
    "dispatchTime": 2
  },
  {
    "id": "enMQks",
    "name": "French Fries",
    "destination": "2111 Sawtelle Blvd, Los Angeles, CA 90025",
    "dispatchTime": 7
  },
  {
    "id": "Gqe5aY",
    "name": "Ice",
    "destination": "454 N Canon Dr, Beverly Hills, CA 90210",
    "dispatchTime": 105
  },
  {
    "id": "AmTjZ4",
    "name": "Carne Asada",
    "destination": "602 Santa Monica Blvd, Santa Monica, CA 90401",
    "dispatchTime": 111
  },
  {
    "id": "xEKoJL",
    "name": "Sherbet",
    "destination": "10426 National Blvd, Los Angeles, CA 90034",
    "dispatchTime": 105
  },
  {
    "id": "jGpfpH",
    "name": "Orange Sorbet",
    "destination": "1129 Glendon Ave, Los Angeles, CA 90024",
    "dispatchTime": 53
  },
  {
    "id": "NPPAPM",
    "name": "Frosty",
    "destination": "1355 Ocean Ave, Santa Monica, CA 90401",
    "dispatchTime": 109
  },
  {
    "id": "BV8i6L",
    "name": "Fresh Bread",
    "destination": "311 N Beverly Dr, Beverly Hills, CA 90210",
    "dispatchTime": 10
  },
  {
    "id": "HnyCoF",
    "name": "Burrito",
    "destination": "52 Windward Ave, Venice, CA 90291",
    "dispatchTime": 66
  },
  {
    "id": "4YtHw9",
    "name": "Icy",
    "destination": "Los Angeles, CA 90012",
    "dispatchTime": 14
  },
  {
    "id": "5kqKRe",
    "name": "Push Pop",
    "destination": "12105 Santa Monica Blvd, Los Angeles, CA 90025",
    "dispatchTime": 73
  },
  {
    "id": "7taEet",
    "name": "Pasta",
    "destination": "3105 Washington Blvd, Marina del Rey, CA 90292",
    "dispatchTime": 83
  },
  {
    "id": "Nb35By",
    "name": "Chicken Nuggets",
    "destination": "13223 W Washington Blvd, Los Angeles, CA 90066",
    "dispatchTime": 51
  },
  {
    "id": "qZTVKh",
    "name": "Ice Cream Sandwich",
    "destination": "2933 Santa Monica Blvd, Santa Monica, CA 90404",
    "dispatchTime": 52
  },
  {
    "id": "UxZKY7",
    "name": "Taco",
    "destination": "2727 Main St, Ste B, Santa Monica, CA 90405",
    "dispatchTime": 24
  },
  {
    "id": "WcEGuX",
    "name": "Tomato Soup",
    "destination": "1240 3rd St, Santa Monica, CA 90401",
    "dispatchTime": 19
  },
  {
    "id": "W3dzYv",
    "name": "Vanilla Ice Cream",
    "destination": "1518 Montana Ave, Santa Monica, CA 90403",
    "dispatchTime": 119
  },
  {
    "id": "zx8fe5",
    "name": "Poppers",
    "destination": "1810 Ocean Ave, Santa Monica, CA 90401",
    "dispatchTime": 88
  },
  {
    "id": "FuMWPP",
    "name": "Popsicle",
    "destination": "1453 Westwood Blvd, Los Angeles, CA 90024",
    "dispatchTime": 83
  },
  {
    "id": "jK6KQJ",
    "name": "Strawberries",
    "destination": "202 Wilshire Blvd, Santa Monica, CA 90401",
    "dispatchTime": 120
  },
  {
    "id": "Knfxmk",
    "name": "Brown Rice",
    "destination": "5407 Sepulveda Blvd, Culver City, CA 90230",
    "dispatchTime": 40
  },
  {
    "id": "77wsMP",
    "name": "Cheese Pizza",
    "destination": "1652 Ocean Ave, Santa Monica, CA 90401",
    "dispatchTime": 63
  },
  {
    "id": "sUfj5Z",
    "name": "Pressed Juice",
    "destination": "307 Lincoln Blvd, Venice, CA 90291",
    "dispatchTime": 108
  },
  {
    "id": "WC4cRh",
    "name": "Coconut",
    "destination": "1360 3rd St Promenade, Santa Monica, CA 90401",
    "dispatchTime": 109
  },
  {
    "id": "YALLuc",
    "name": "Onion Rings",
    "destination": "4th St & Adelaide Dr, Santa Monica, CA 90402",
    "dispatchTime": 26
  },
  {
    "id": "hGQgjK",
    "name": "Fish Tacos",
    "destination": "11918 Washington Blvd, Los Angeles, CA 90066",
    "dispatchTime": 57
  },
  {
    "id": "Nk4V4P",
    "name": "Pot Stickers",
    "destination": "11740 San Vicente Blvd, Fl 2, Brentwood, CA 90049",
    "dispatchTime": 70
  },
  {
    "id": "qcUKfu",
    "name": "Kombucha",
    "destination": "122 S Beverly Dr, Beverly Hills, CA 90212",
    "dispatchTime": 80
  },
  {
    "id": "xJRSth",
    "name": "Mixed Greens",
    "destination": "233 S Beverly Dr, Beverly Hills, CA 90212",
    "dispatchTime": 85
  },
  {
    "id": "zdTnpg",
    "name": "Sushi",
    "destination": "10922 Kinross Ave, Los Angeles, CA 90024",
    "dispatchTime": 72
  },
  {
    "id": "XhhFwK",
    "name": "Apples",
    "destination": "The Seychelle, 1755 Ocean Ave, Santa Monica, CA 90401",
    "dispatchTime": 79
  },
  {
    "id": "9GQLoA",
    "name": "Kebab",
    "destination": "1442 2nd St, Santa Monica, CA 90401",
    "dispatchTime": 71
  },
  {
    "id": "A7f5aq",
    "name": "Mac & Cheese",
    "destination": "12924 Washington Blvd, Culver City, CA 90066",
    "dispatchTime": 90
  },
  {
    "id": "ouDf8Q",
    "name": "Corn Dog",
    "destination": "1916 Westwood Blvd, Los Angeles, CA 90025",
    "dispatchTime": 87
  },
  {
    "id": "Fet74W",
    "name": "Grilled Corn Salad",
    "destination": "1519 Wilshire Blvd, Santa Monica, CA 90403",
    "dispatchTime": 73
  },
  {
    "id": "mrAi4b",
    "name": "Pistachio Ice Cream",
    "destination": "2628 Wilshire Blvd, Santa Monica, CA 90403",
    "dispatchTime": 41
  },
  {
    "id": "JAHtHi",
    "name": "Strawberyy Banana Split",
    "destination": "1211 Montana Ave, Santa Monica, CA 90403",
    "dispatchTime": 59
  },
  {
    "id": "sqW2Km",
    "name": "McFlury",
    "destination": "2654 Main St, Santa Monica, CA 90405",
    "dispatchTime": 13
  },
  {
    "id": "iqTeqb",
    "name": "Acai Bowl",
    "destination": "2622 Pico Blvd, Santa Monica, CA 90405",
    "dispatchTime": 34
  },
  {
    "id": "c3mvrw",
    "name": "Yogurt",
    "destination": "12746-3 W Jefferson Blvd, Ste 3140, Playa Vista, CA 90094",
    "dispatchTime": 111
  },
  {
    "id": "TsD9JN",
    "name": "Chocolate Gelato",
    "destination": "4507 S Centinela Ave, Los Angeles, CA 90066",
    "dispatchTime": 115
  },
  {
    "id": "XgCaSL",
    "name": "Cobb Salad",
    "destination": "3360 Ocean Park Blvd, Santa Monica, CA 90405",
    "dispatchTime": 67
  },
  {
    "id": "JDdMzi",
    "name": "Cottage Cheese",
    "destination": "9748 Washington Blvd, Culver City, CA 90232",
    "dispatchTime": 74
  },
  {
    "id": "KeooUN",
    "name": "Coke",
    "destination": "11651 Santa Monica Blvd, Los Angeles, CA 90025",
    "dispatchTime": 77
  },
  {
    "id": "3NmUbb",
    "name": "Snow Cone",
    "destination": "364 N Beverly Dr, Beverly Hills, CA 90210",
    "dispatchTime": 20
  },
  {
    "id": "jJUzCE",
    "name": "Pad See Ew",
    "destination": "1414 Lincoln Blvd, Santa Monica, CA 90401",
    "dispatchTime": 63
  },
  {
    "id": "fZSmDa",
    "name": "Chunky Monkey",
    "destination": "2553 Lincoln Blvd, Venice, CA 90291",
    "dispatchTime": 47
  },
  {
    "id": "C73cMB",
    "name": "Beef Stew",
    "destination": "8406 Pershing Dr, Playa Del Rey, CA 90293",
    "dispatchTime": 96
  },
  {
    "id": "AKi9ti",
    "name": "Cheese",
    "destination": "6220 W 87th St, Los Angeles, CA 90045",
    "dispatchTime": 60
  },
  {
    "id": "wzV4rq",
    "name": "Spinach Omelet",
    "destination": "301 N Canon Dr, Beverly Hills, CA 90210",
    "dispatchTime": 67
  },
  {
    "id": "QWwrJS",
    "name": "Beef Hash",
    "destination": "9226 W Pico Blvd, Los Angeles, CA 90035",
    "dispatchTime": 113
  },
  {
    "id": "QmMFKs",
    "name": "Pork Chop",
    "destination": "6000 Sepulveda Blvd, Culver City, CA 90230",
    "dispatchTime": 22
  },
  {
    "id": "GRcjpu",
    "name": "Kale Salad",
    "destination": "219 Arizona Ave, Santa Monica, CA 90401",
    "dispatchTime": 11
  },
  {
    "id": "7ZNRK9",
    "name": "Fresh Fruit",
    "destination": "3021 Lincoln Blvd, Santa Monica, CA 90405",
    "dispatchTime": 70
  },
  {
    "id": "zuTsXE",
    "name": "Cranberry Salad",
    "destination": "4065 Jackson Ave, Culver City, CA 90232",
    "dispatchTime": 11
  },
  {
    "id": "2zBRsz",
    "name": "Fudge Ice Cream Cake",
    "destination": "6081 Center Dr, Ste 203, Los Angeles, CA 90045",
    "dispatchTime": 28
  },
  {
    "id": "4kWeLz",
    "name": "Mint Chocolate Ice Cream",
    "destination": "301 Santa Monica Pier, Bldg 9, Santa Monica, CA 90401",
    "dispatchTime": 101
  },
  {
    "id": "gEdkvF",
    "name": "Vegan Pizza",
    "destination": "11951 Santa Monica Blvd, Los Angeles, CA 90025",
    "dispatchTime": 1
  },
  {
    "id": "3CGPQZ",
    "name": "Orange Chicken",
    "destination": "2127 Lincoln Blvd, Santa Monica, CA 90405",
    "dispatchTime": 83
  },
  {
    "id": "DeLjgD",
    "name": "MeatLoaf",
    "destination": "4213 Admiralty Way, Killer Shrimp restaurant, Marina Del Rey, CA 90292",
    "dispatchTime": 111
  },
  {
    "id": "eppU4V",
    "name": "Milk",
    "destination": "1900 S Lincoln Blvd, Los Angeles, CA 90291",
    "dispatchTime": 40
  },
  {
    "id": "4RjiCK",
    "name": "Pastrami Sandwich",
    "destination": "930 Hilgard Ave, Los Angeles, CA 90024",
    "dispatchTime": 13
  },
  {
    "id": "4ftcQP",
    "name": "Arugula",
    "destination": "1629 Wilshire Blvd, Santa Monica, CA 90403",
    "dispatchTime": 14
  },
  {
    "id": "QCYxV2",
    "name": "Pickles",
    "destination": "1811 Pico Blvd, Santa Monica, CA 90405",
    "dispatchTime": 103
  },
  {
    "id": "QJpuNd",
    "name": "Chicken",
    "destination": "8732 S Sepulveda Blvd, Los Angeles, CA 90045",
    "dispatchTime": 119
  },
  {
    "id": "XCZQ2S",
    "name": "Cookie Dough",
    "destination": "2805 Abbot Kinney Blvd, Venice, CA 90291",
    "dispatchTime": 106
  },
  {
    "id": "uKcXNd",
    "name": "Hamburger",
    "destination": "3117 Ocean Park Blvd, Santa Monica, CA 90405",
    "dispatchTime": 62
  },
  {
    "id": "uLQfc2",
    "name": "French Fries",
    "destination": "12150 Millennium Dr, Ste 101, Los Angeles, CA 90094",
    "dispatchTime": 18
  },
  {
    "id": "svdMhv",
    "name": "Ice",
    "destination": "235 Main St, Venice, CA 90291",
    "dispatchTime": 43
  },
  {
    "id": "gZodbS",
    "name": "Carne Asada",
    "destination": "1915 Westwood Blvd, Los Angeles, CA 90025",
    "dispatchTime": 48
  },
  {
    "id": "4E253Z",
    "name": "Sherbet",
    "destination": "10942 Weyburn Ave, Los Angeles, CA 90024",
    "dispatchTime": 64
  },
  {
    "id": "2ujead",
    "name": "Orange Sorbet",
    "destination": "11819 Wilshire Blvd, Ste 106B, Los Angeles, CA 90025",
    "dispatchTime": 52
  },
  {
    "id": "LnjCGs",
    "name": "Frosty",
    "destination": "208 N Rodeo Dr, Beverly Hills, CA 90210",
    "dispatchTime": 18
  },
  {
    "id": "i9rhk3",
    "name": "Fresh Bread",
    "destination": "820 Wilshire Blvd, Santa Monica, CA 90401",
    "dispatchTime": 72
  },
  {
    "id": "qnMcX9",
    "name": "Burrito",
    "destination": "1615 Ocean Front Walk, Santa Monica, CA 90401",
    "dispatchTime": 67
  },
  {
    "id": "5dV4js",
    "name": "Icy",
    "destination": "12012 W Pico Blvd, Los Angeles, CA 90064",
    "dispatchTime": 57
  },
  {
    "id": "C3LJF6",
    "name": "Push Pop",
    "destination": "1942 Westwood Blvd, Los Angeles, CA 90025",
    "dispatchTime": 61
  },
  {
    "id": "tHgCD7",
    "name": "Pasta",
    "destination": "13723 Fiji Way, Marina Del Rey, CA 90292",
    "dispatchTime": 114
  },
  {
    "id": "jYbwMo",
    "name": "Chicken Nuggets",
    "destination": "80 Windward Ave, Venice, CA 90291",
    "dispatchTime": 110
  },
  {
    "id": "p4PYwE",
    "name": "Ice Cream Sandwich",
    "destination": "2709 Main St, Santa Monica, CA 90405",
    "dispatchTime": 49
  },
  {
    "id": "JzUxmX",
    "name": "Taco",
    "destination": "Carroll Ct & Eastern Canal Ct, Venice, CA 90293",
    "dispatchTime": 15
  },
  {
    "id": "Zqotpo",
    "name": "Tomato Soup",
    "destination": "3857 Overland Ave, Culver City, CA 90232",
    "dispatchTime": 95
  },
  {
    "id": "C588Fu",
    "name": "Vanilla Ice Cream",
    "destination": "405 Washington Blvd, Marina Del Rey, CA 90292",
    "dispatchTime": 38
  },
  {
    "id": "yVoJDX",
    "name": "Poppers",
    "destination": "826 Pico Blvd, Santa Monica, CA 90405",
    "dispatchTime": 108
  },
  {
    "id": "boPpCZ",
    "name": "Popsicle",
    "destination": "1200 Getty Center Dr, Los Angeles, CA 90049",
    "dispatchTime": 61
  },
  {
    "id": "bugu5R",
    "name": "Strawberries",
    "destination": "500 World Way Terminal 5, Los Angeles, CA 90045",
    "dispatchTime": 107
  },
  {
    "id": "fccsoP",
    "name": "Brown Rice",
    "destination": "11835 Wilshire Blvd, Los Angeles, CA 90024",
    "dispatchTime": 58
  },
  {
    "id": "rDf73P",
    "name": "Cheese Pizza",
    "destination": "9562 W Pico Blvd, Los Angeles, CA 90035",
    "dispatchTime": 48
  },
  {
    "id": "z96ozm",
    "name": "Pressed Juice",
    "destination": "12565 Washington Blvd, Los Angeles, CA 90066",
    "dispatchTime": 65
  },
  {
    "id": "uGUP8q",
    "name": "Coconut",
    "destination": "1910 Wilshire Blvd, Santa Monica, CA 90403",
    "dispatchTime": 37
  },
  {
    "id": "mRbjVB",
    "name": "Onion Rings",
    "destination": "311 Washington Blvd, Marina Del Rey, CA 90292",
    "dispatchTime": 101
  },
  {
    "id": "87khM8",
    "name": "Fish Tacos",
    "destination": "1725 Broadway, Santa Monica, CA 90404",
    "dispatchTime": 47
  },
  {
    "id": "8RCDgs",
    "name": "Pot Stickers",
    "destination": "1028 Wilshire Blvd, Santa Monica, CA 90401",
    "dispatchTime": 110
  },
  {
    "id": "Yn448N",
    "name": "Kombucha",
    "destination": "1412 3rd St Promenade, Santa Monica, CA 90401",
    "dispatchTime": 9
  },
  {
    "id": "3M6VYf",
    "name": "Mixed Greens",
    "destination": "11057 Santa Monica Blvd, Los Angeles, CA 90025",
    "dispatchTime": 116
  },
  {
    "id": "SncVtB",
    "name": "Sushi",
    "destination": "3819 Main St, Culver City, CA 90232",
    "dispatchTime": 44
  },
  {
    "id": "6o3Jga",
    "name": "Apples",
    "destination": "10250 Santa Monica Blvd, Los Angeles, CA 90067",
    "dispatchTime": 93
  },
  {
    "id": "KtqYiA",
    "name": "Kebab",
    "destination": "10845 Lindbrook Dr, Los Angeles, CA 90024",
    "dispatchTime": 108
  },
  {
    "id": "LrStHs",
    "name": "Mac & Cheese",
    "destination": "447 Canon Dr, Beverly Hills, CA 90210",
    "dispatchTime": 93
  },
  {
    "id": "JNUBA5",
    "name": "Corn Dog",
    "destination": "1 World Way, Terminal 5, Los Angeles International Airport, Los Angeles, CA 90045",
    "dispatchTime": 39
  },
  {
    "id": "TCKXJe",
    "name": "Grilled Corn Salad",
    "destination": "10 W Century Dr, Los Angeles, CA 90067",
    "dispatchTime": 83
  },
  {
    "id": "yuuHhk",
    "name": "Pistachio Ice Cream",
    "destination": "1777 Westwood Blvd, Los Angeles, CA 90024",
    "dispatchTime": 111
  }
]
