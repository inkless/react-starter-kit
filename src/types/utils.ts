export interface StringTMap<T> {
  [key: string]: T;
}
export interface NumberTMap<T> {
  [key: number]: T;
}

export interface StringAnyMap extends StringTMap<any> {}
export interface NumberAnyMap extends NumberTMap<any> {}

export interface StringStringMap extends StringTMap<string> {}
export interface NumberStringMap extends NumberTMap<string> {}

export interface StringNumberMap extends StringTMap<number> {}
export interface NumberNumberMap extends NumberTMap<number> {}

export interface StringBooleanMap extends StringTMap<boolean> {}
export interface NumberBooleanMap extends NumberTMap<boolean> {}

export type valueof<T> = T[keyof T];

/**
 * Omit some properties in one type.
 *
 * e.g.
 * ```
 * type Foo = { apple: string; banana: string };
 * type Bar = Omit<Foo, 'apple'>;
 * ```
 * so `Bar` is equivalant to `{ banana: string }`
 */
export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
