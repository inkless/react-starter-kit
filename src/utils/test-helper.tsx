import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import { theme } from '@src/theme';
import { createRootReducer } from '@src/store/reducers';

export function renderWithTheme(tree: JSX.Element, options?: any) {
  return render(tree, {
    ...options,
    context: { theme },
  });
}

export function shallowWithTheme(tree: JSX.Element, options?: any) {
  return shallow(tree, {
    ...options,
    context: { theme },
  });
}

export function mountWithTheme(tree: JSX.Element, options?: any) {
  return mount(<ThemeProvider theme={theme}>{tree}</ThemeProvider>, options);
}

export function createInitialState() {
  const rootReducer = createRootReducer({} as any);
  return rootReducer(undefined, '' as any);
}
