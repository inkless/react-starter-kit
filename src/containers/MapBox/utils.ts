import request from '@src/utils/request';
import { lineString, Feature } from '@turf/helpers';
import turfLength from '@turf/length';
import { decode } from '@mapbox/polyline';

import { StringAnyMap } from '@src/types/utils';
import { MAPBOX_API_URL } from './constants';

const ACCESS_TOKEN = process.env.MapboxAccessToken;
const POLYLINE_PRECISION = 5;
const SECONDS_IN_HOUR = 3600;
const SPEED_BASE = 20;

/**
 * Fetch driving route from origin to destination
 *
 * @param origin
 * @param destination
 */
export async function fetchDrivingRoute(origin: [number, number], destination: [number, number]) {
  const baseUrl = `${MAPBOX_API_URL}/directions/v5/mapbox/driving-traffic`;
  const coordinates = `${origin.join(',')};${destination.join(',')}`;
  const url = `${baseUrl}/${coordinates}?steps=true&overview=full&access_token=${ACCESS_TOKEN}`;
  const data = await request(url);
  return data.routes[0];
}

/**
 * Given any address, find the geocoding
 * @param address
 */
export async function fetchGeocoding(address: string) {
  const url = `${MAPBOX_API_URL}/geocoding/v5/mapbox.places/${encodeURIComponent(
    address
  )}.json?access_token=${ACCESS_TOKEN}`;
  const data = await request(url);
  return data.features[0].center;
}

/**
 * Generate a speed between 20 to 25 miles per hour
 * @returns miles per second
 */
export function generateRandomSpeed() {
  return (Math.random() * 5 + SPEED_BASE) / SECONDS_IN_HOUR;
}

/**
 * Decode geometry and return geojson route
 * @param geometry
 * @returns geojson lineString
 */
export function createRouteFromGeometry(geometry: string, options?: StringAnyMap) {
  const decoded = decode(geometry, POLYLINE_PRECISION).map(c => {
    return c.reverse();
  });

  return lineString(decoded, options);
}

/**
 * Given a route, calculate the distance of it
 * @param route
 * @returns distance in miles
 */
export function calculateRouteLength(route: Feature) {
  return turfLength(route, { units: 'miles' });
}
