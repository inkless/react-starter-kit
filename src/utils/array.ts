import { StringAnyMap, StringTMap } from '@src/types/utils';

export function groupBy<T>(list: T[], func: (item: T) => string | boolean): StringTMap<T[]> {
  return list.reduce(
    (accumulate, item) => {
      const key = String(func(item));
      (accumulate[key] = accumulate[key] || []).push(item);
      return accumulate;
    },
    {} as StringAnyMap
  );
}
