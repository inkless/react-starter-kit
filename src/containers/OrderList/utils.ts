import { groupBy } from '@src/utils/array';
import { PendingOrder } from './types';

/**
 * reduce all orders' dispatchTime by 1 second,
 * move them to delivering if dispatchTime is 0
 *
 * @param orders - pending orders
 * @returns
 */
export function tickOneSecond(orders: PendingOrder[]) {
  const updatedOrders = orders.map(order => ({
    ...order,
    dispatchTime: order.dispatchTime - 1,
  }));

  const { true: delivering = [], false: pending = [] } = groupBy(
    updatedOrders,
    ({ dispatchTime }) => dispatchTime === 0
  );

  return {
    delivering,
    pending,
  };
}
