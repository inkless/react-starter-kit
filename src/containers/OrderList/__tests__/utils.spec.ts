import { pending } from '@src/components/OrderList/__tests__/__fixture__/orders';
import { tickOneSecond } from '../utils';

describe('OrderList Utils', () => {
  describe('#tickOneSecond', () => {
    it('should update dispatchTime correctly', () => {
      const updatedOrders = tickOneSecond(pending);
      expect(updatedOrders).toMatchSnapshot();
    });
  });
});
