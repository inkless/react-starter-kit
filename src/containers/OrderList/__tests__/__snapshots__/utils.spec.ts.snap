// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList Utils #tickOneSecond should update dispatchTime correctly 1`] = `
Object {
  "delivering": Array [],
  "pending": Array [
    Object {
      "destination": "185 Culver Blvd, Playa Del Rey, CA 90293",
      "dispatchTime": 118,
      "id": "9H4whD",
      "name": "Banana Split",
    },
    Object {
      "destination": "13455 Maxella Ave, Marina Del Rey, CA 90292",
      "dispatchTime": 97,
      "id": "EUtSqE",
      "name": "McFlury",
    },
    Object {
      "destination": "4720 Admiralty Way, Marina Del Rey, CA 90292",
      "dispatchTime": 34,
      "id": "uoierx",
      "name": "Acai Bowl",
    },
    Object {
      "destination": "220 Rose Ave, Venice, CA 90291",
      "dispatchTime": 104,
      "id": "oP6QHd",
      "name": "Yogurt",
    },
    Object {
      "destination": "2323 Wilshire Blvd, Santa Monica, CA 90403",
      "dispatchTime": 72,
      "id": "aS4EFD",
      "name": "Chocolate Gelato",
    },
    Object {
      "destination": "1630 Ocean Park Blvd, Santa Monica, CA 90405",
      "dispatchTime": 12,
      "id": "4WVKZu",
      "name": "Cobb Salad",
    },
    Object {
      "destination": "1416 4th St, Santa Monica, CA 90401",
      "dispatchTime": 2,
      "id": "5sSNvC",
      "name": "Cottage Cheese",
    },
    Object {
      "destination": "8115 W Manchester Ave, Playa Del Rey, CA 90293",
      "dispatchTime": 118,
      "id": "Fv8AdV",
      "name": "Coke",
    },
    Object {
      "destination": "2055 Sawtelle Blvd, Los Angeles, CA 90025",
      "dispatchTime": 48,
      "id": "WFG4B3",
      "name": "Snow Cone",
    },
    Object {
      "destination": "2906 Main St, Santa Monica, CA 90405",
      "dispatchTime": 93,
      "id": "woqBZv",
      "name": "Pad See Ew",
    },
  ],
}
`;
